File: web/src/api/orderfood/ai.js
import { orderFoodRequest } from './request'

const encodePath = (value) => encodeURIComponent(value)

// 平台整体能力策略
export const getPlatformCapabilityPolicy = () =>
  orderFoodRequest({
    path: '/platform-capability-policy',
    method: 'get'
  })

export const updatePlatformCapabilityPolicy = (data) =>
  orderFoodRequest({
    path: '/platform-capability-policy',
    method: 'put',
    data,
    mutation: true
  })

export const updatePlatformCapabilityEmergencyStatus = (data) =>
  orderFoodRequest({
    path: '/platform-capability-policy/emergency-status',
    method: 'put',
    data,
    mutation: true
  })

// AI 供应商
export const getAIProviderList = (params) =>
  orderFoodRequest({
    path: '/ai-providers',
    method: 'get',
    params
  })

export const getAIProviderDetail = (providerId) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}`,
    method: 'get'
  })

export const getAIProviderModelOptions = (providerId) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}/models`,
    method: 'get'
  })

export const getAIModelProviderOptions = () =>
  orderFoodRequest({
    path: '/ai-model-provider-options',
    method: 'get'
  })

export const createAIProvider = (data) =>
  orderFoodRequest({
    path: '/ai-providers',
    method: 'post',
    data,
    mutation: true
  })

export const updateAIProvider = (providerId, data) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}`,
    method: 'put',
    data,
    mutation: true
  })

export const updateAIProviderStatus = (providerId, data) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}/status`,
    method: 'put',
    data,
    mutation: true
  })

export const deleteAIProvider = (providerId, data) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}`,
    method: 'delete',
    data,
    mutation: true
  })

export const testAIProviderConnection = (providerId, data) =>
  orderFoodRequest({
    path: `/ai-providers/${encodePath(providerId)}/connection-tests`,
    method: 'post',
    data,
    mutation: true
  })

// AI 模型
export const getAIModelList = (params) =>
  orderFoodRequest({
    path: '/ai-models',
    method: 'get',
    params
  })

export const getAIModelDetail = (modelId) =>
  orderFoodRequest({
    path: `/ai-models/${encodePath(modelId)}`,
    method: 'get'
  })

export const createAIModel = (data) =>
  orderFoodRequest({
    path: '/ai-models',
    method: 'post',
    data,
    mutation: true
  })

export const updateAIModel = (modelId, data) =>
  orderFoodRequest({
    path: `/ai-models/${encodePath(modelId)}`,
    method: 'put',
    data,
    mutation: true
  })

export const updateAIModelStatus = (modelId, data) =>
  orderFoodRequest({
    path: `/ai-models/${encodePath(modelId)}/status`,
    method: 'put',
    data,
    mutation: true
  })

export const deleteAIModel = (modelId, data) =>
  orderFoodRequest({
    path: `/ai-models/${encodePath(modelId)}`,
    method: 'delete',
    data,
    mutation: true
  })

// 六项 AI 执行能力与提示词
export const getAICapabilityList = (params) =>
  orderFoodRequest({
    path: '/ai-capabilities',
    method: 'get',
    params
  })

export const getAICapabilityDetail = (capabilityCode) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}`,
    method: 'get'
  })

export const updateAICapability = (capabilityCode, data) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}`,
    method: 'put',
    data,
    mutation: true
  })

export const getAICapabilityPromptWorkspace = (capabilityCode) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}/prompts`,
    method: 'get'
  })

export const updateAICapabilityPrompt = (capabilityCode, data) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}/prompts`,
    method: 'put',
    data,
    mutation: true
  })

export const validateAICapabilityPrompt = (capabilityCode, data) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}/prompt-validation`,
    method: 'post',
    data,
    mutation: true
  })

export const renderAICapabilityPrompt = (capabilityCode, data) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}/prompt-render-preview`,
    method: 'post',
    data,
    mutation: true
  })

export const testAICapabilityPrompt = (capabilityCode, data) =>
  orderFoodRequest({
    path: `/ai-capabilities/${encodePath(capabilityCode)}/prompt-tests`,
    method: 'post',
    data,
    mutation: true
  })

File: server/router/ai/ai.go
package ai

import "github.com/gin-gonic/gin"

type AIRouter struct{}

// InitAIRouter registers internal Gin paths under /orderfood. The web client
// sees the frozen external /api/orderfood prefix through the existing proxy.
func (router *AIRouter) InitAIRouter(orderFoodGroup *gin.RouterGroup) {
	group := orderFoodGroup
	{
		group.GET("/platform-capability-policy", aiApi.GetPlatformPolicy)
		group.PUT("/platform-capability-policy", aiApi.UpdatePlatformPolicy)
		group.PUT("/platform-capability-policy/emergency-status", aiApi.SetPlatformEmergencyStatus)

		group.GET("/ai-providers", aiApi.ListAIProviders)
		group.GET("/ai-providers/:providerId", aiApi.GetAIProvider)
		group.GET("/ai-providers/:providerId/models", aiApi.ListAIProviderModels)
		group.GET("/ai-model-provider-options", aiApi.ListAIModelProviders)
		group.POST("/ai-providers", aiApi.CreateAIProvider)
		group.PUT("/ai-providers/:providerId", aiApi.UpdateAIProvider)
		group.PUT("/ai-providers/:providerId/status", aiApi.UpdateAIProviderStatus)
		group.DELETE("/ai-providers/:providerId", aiApi.DeleteAIProvider)
		group.POST("/ai-providers/:providerId/connection-tests", aiApi.TestAIProviderConnection)

		group.GET("/ai-models", aiApi.ListAIModels)
		group.GET("/ai-models/:modelId", aiApi.GetAIModel)
		group.POST("/ai-models", aiApi.CreateAIModel)
		group.PUT("/ai-models/:modelId", aiApi.UpdateAIModel)
		group.PUT("/ai-models/:modelId/status", aiApi.UpdateAIModelStatus)
		group.DELETE("/ai-models/:modelId", aiApi.DeleteAIModel)

		group.GET("/ai-capabilities", aiApi.ListAICapabilities)
		group.GET("/ai-capabilities/:capabilityCode", aiApi.GetAICapability)
		group.PUT("/ai-capabilities/:capabilityCode", aiApi.UpdateAICapability)
		group.GET("/ai-capabilities/:capabilityCode/prompts", aiApi.GetAIPromptWorkspace)
		group.PUT("/ai-capabilities/:capabilityCode/prompts", aiApi.UpdateAIPrompt)
		group.POST("/ai-capabilities/:capabilityCode/prompt-validation", aiApi.ValidateAIPrompt)
		group.POST("/ai-capabilities/:capabilityCode/prompt-render-preview", aiApi.RenderAIPrompt)
		group.POST("/ai-capabilities/:capabilityCode/prompt-tests", aiApi.TestAIPrompt)
	}
}
